src/openalea/lpy/parameters/schema/material.json

```diff
@@ -0,0 +1,49 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "material",
+    "type": "object",
+    "properties": {
+        "name": {
+            "type": "string",
+            "pattern": "^[^\\d\\W]\\w*\\Z"
+        },
+        "id": {
+            "type": "integer",
+            "minimum": 0
+        },
+        "type": {
+            "enum": ["Material"]
+        },
+        "index": {
+            "type": "integer",
+            "minimum": 0
+        },
+        "ambient": {
+            "$ref": "color.json"
+        },
+        "specular": {
+            "$ref": "color.json"
+        },
+        "emission": {
+            "$ref": "color.json"
+        },
+        "diffuse": {
+            "type": "number",
+            "minimum": 0,
+            "maximum": 3
+        },
+        "transparency": {
+            "type": "number",
+            "minimum": 0,
+            "maximum": 1
+        },
+        "shininess": {
+            "type": "number",
+            "minimum": 0,
+            "maximum": 1
+        }
+    },
+    "required": ["name", "index", "ambient"],
```

Comment:
"type" is not required but parsing a dict fails with KeyError: 'type'

```py
lp = LsystemParameters()
s = '''{
    "schema": "lpy",
    "version": "1.0",
    "options": {},
    "materials": [
            {
            "name": "trunk",
            "index": 1,
            "ambient": [
                91,
                69,
                52
            ],
            "diffuse": 1.0,
            "specular": [
                40,
                40,
                40
            ],
            "emission": [
                0,
                0,
                0
            ],
            "transparency": 0.0,
            "shininess": 0.2
        }
    ],
    "parameters": [],
    "credits": {}
}'''
p = json.loads(s)

print(lp.is_valid_schema(p))
lp.loads(s)
```

Reply:
I put the type as mandatory field.